src/parser.rs

```diff
@@ -49,18 +38,13 @@ impl ContextValues {
     }
 }
 
-pub fn parse(input: &'static str, locale: &'static str, zero_indexing: bool) -> Tokenizer {
+pub fn parse(input: &'static str,  zero_indexing: bool) -> Tokenizer {
```

Comment:
Why does the `input` argument have `static` lifetime? Is it because you only pass the hardcodded string to it? This definitely not the case, this should work for any valid lifetime.